hydrolib/core/io/rr/network/serializer.py

```diff
@@ -0,0 +1,34 @@
+import os
+from pathlib import Path
+
+
+class NodeFileSerializer:
+    @staticmethod
+    def serialize(path: Path, data: dict):
+        """
+        Serializes the RR node topology data to the file at the specified path.
+
+        Attributes:
+            path (Path): The path to the destination file.
+            data (Dict): The data to be serialized.
+        """
+
+        path.parent.mkdir(parents=True, exist_ok=True)
+
+        with path.open("wb") as f:
+            for node in data["node"]:
+                line = f"NODE {NodeFileSerializer._to_line(node)} node{os.linesep}"
```

Comment:
Not super important, but why not add the "NODE..node" also to the function ```_to_line()```?

Reply:
I had been questioning this but eventually decided on the current solution.
I figured that the `_to_line(node)` method would convert a node to the data (in a string).
And this data should then be enclosed between the two keywords. 
But it does not really matter to me as well :)